From 30210fd8104c499f1300700547894f83cc3b0f6d Mon Sep 17 00:00:00 2001
From: Long Huynh Huu <long@putputlaw.de>
Date: Thu, 20 Aug 2026 23:19:33 +0200
Subject: [PATCH] fix: stale symlinks into nix store when using nix

---
 lua/nvim-treesitter/install.lua | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/lua/nvim-treesitter/install.lua b/lua/nvim-treesitter/install.lua
index 71d5a3c7..e41b656b 100644
--- a/lua/nvim-treesitter/install.lua
+++ b/lua/nvim-treesitter/install.lua
@@ -435,14 +435,14 @@ local function try_install_lang(lang, cache_dir, install_dir, generate)
     local query_dir = fs.joinpath(config.get_install_dir('queries'), lang)
     local task ---@type function
 
-    if repo and repo.queries and repo.path then -- link queries from local repo
+    if repo and repo.queries and repo.path then -- copy queries from local repo
       query_src = fs.joinpath(fs.normalize(repo.path), repo.queries)
-      task = do_link_queries
+      task = do_copy_queries
     elseif repo and repo.queries then -- copy queries from tarball
       query_src = fs.joinpath(cache_dir, project_name, repo.queries)
       task = do_copy_queries
-    elseif uv.fs_stat(query_src) then -- link queries from runtime
-      task = do_link_queries
+    elseif uv.fs_stat(query_src) then -- copy queries from runtime
+      task = do_copy_queries
     end
 
     if task then
-- 
2.54.0

